CDLUC3/dmsp_frontend_prototype
# Made token handling updates in the frontend to accomodate recent backend changes regarding access, refresh, and csrf tokens

## Description

This work was done in response to the work recently done [here](https://github.com/CDLUC3/dmsp_backend_prototype/pull/90) on the backend to add access and refresh tokens to HTTP-only cookies on the client browser, and to add the use of csrf tokens.

- Removed API endpoints for /get-token, /logout and /set-cookie, since the backend is now adding auth cookies, using its endpoint to log out the user and remove cookies, and verifying tokens on the back end
- Created a CsrfContext to track the state of the csrf token
-Updated /login and /signup pages to pass the csrf token in the header to the backend. Created a new, shared errorhandler for both pages to handle error responses
- Fixed a bug with /components/Editor/index.tsx page
- Updated Header component to call the new backend /apollo-signout endpoint when clicking the "Logout" button
- Update graphql errorLink and added unit test for it
- Added cypress tests for /login and /signup
- Updated README.md

Fixes # ([116](https://github.com/CDLUC3/dmsp_frontend_prototype/issues/116))

## Type of change

## How Has This Been Tested?

- Added and updated unit tests.
- Ran cypress testing (both headed and headless) for new tests
- Currently the refreshing of the access token is not working. The backend has another ticket to fix this. Also, graphql requests for some data, like "Sections" is not working. That is also being addressed as part of this ticket: [#111](https://github.com/CDLUC3/dmsp_backend_prototype/issues/111)

/login
1. Start backend development branch
2. Start frontend app, and go to /login page
3. Enter existing username and password, such as for super@example.com
4. After login, note that the `dmspt` and `dmspr` cookies are created

/signup
1. Go to /signup page and enter email and password that comply with requirements
2. Submit. Note after submitting form, that the `dmspt` and `dmspr` cookies have been added

/logout
1. Click the "Logout" button after signing in.
2. Note that the `dmspt` and `dmspr` cookies are removed and the user is redirected to /login page.
cypress

1. Run the `npm run cypress:open` command in your frontend directory. Select which test you want to run for /login, /logout, or /signup.
3. Run the `npm run cypress:run` command in your frontend directory. This will run the headless version of all available cypress tests.

## Checklist:
- [NA] I have completed manual or automated accessibility testing for my changes
- [] Any dependent changes have been merged and published in downstream modules (Will need to merge my backend changes [here](https://github.com/CDLUC3/dmsp_backend_prototype/tree/bug/fix-cors-and-other-small-issues)  before I can merge this.
